aten/src/ATen/core/SingletonSymNodeImpl.h

```diff
@@ -34,12 +35,40 @@ namespace c10 {
 // During tracing the strides of the outputs need to be a function of the size
 // and strides of the inputs so it is important that SingletonSymNode itself is
 // able to express this.
+//
+// NOTE [ SingletonVariant ]
+//
+// Currently, if SingletonSymNodeType::CPP is passed, that means that the
+// singleton is only meant to be used to ferry nested_tensor_size metadata
+// from forward to use in backward. In this case we set `val`, `coeff` etc
+// to bogus values and make sure to error if they are accessed.
+enum class SingletonVariant { PYTHON, CPP };
```

Comment:
I want to see a clearer split here, where CPP is clearly only to be used with C++ strided nested tensor and Python jagged nested tensor.